retrofit2/src/main/java/com/linecorp/armeria/client/http/retrofit2/ArmeriaRetrofitBuilder.java

```diff
@@ -0,0 +1,200 @@
+import retrofit2.Call;
+import retrofit2.CallAdapter;
+import retrofit2.Callback;
+import retrofit2.Converter;
+import retrofit2.Retrofit;
+import retrofit2.Retrofit.Builder;
+
+/**
+ * A helper class for creating a new {@link Retrofit} instance with {@link ArmeriaCallFactory}.
+ * For example,
+ * <pre>{@code
+ *
+ * Retrofit retrofit = new ArmeriaRetrofitBuilder("none+http://localhost:8080/")
+ *     .build();
+ *
+ * MyApi api = retrofit.create(MyApi.class);
+ * Response&lt;User&gt; user = api.getUser().execute();
+ * }
+ * </pre>
+ *
+ * <p>ArmeriaRetrofitBuilder even supports EndpointGroup, so you can create Retrofit like below,
+ * <pre>{@code
+ *
+ * EndpointGroupRegistry.register("foo",
+ *                                new StaticEndpointGroup(Endpoint.of("127.0.0.1", 8080)),
+ *                                ROUND_ROBIN);
+ *
+ * Retrofit retrofit = new ArmeriaRetrofitBuilder("none+http://group:foo/")
+ *     .build();
+ * }
+ * </pre>
+ */
+public final class ArmeriaRetrofitBuilder {
+
+    private static final Pattern GROUP_PREFIX_PATTERN = Pattern.compile("^[_0-9a-z]+$");
+    private static final Function<? super ClientOptions, ClientOptions> DEFAULT_CONFIGURATOR =
+            Function.identity();
+    private static final String SLASH = "/";
+
+    private final Retrofit.Builder retrofitBuilder;
+    private final ClientFactory clientFactory;
+    private String baseUrl;
+    private String basePath;
+    private Function<? super ClientOptions, ClientOptions> configurator = DEFAULT_CONFIGURATOR;
+
+    /**
+     * Creates a {@link ArmeriaRetrofitBuilder} with default {@link ClientFactory}.
+     */
+    public ArmeriaRetrofitBuilder() {
+        this(ClientFactory.DEFAULT);
+    }
+
+    /**
+     * Creates a {@link ArmeriaRetrofitBuilder} with {@link ClientFactory}.
+     */
+    public ArmeriaRetrofitBuilder(ClientFactory factory) {
+        clientFactory = factory;
+        retrofitBuilder = new Retrofit.Builder();
+    }
+
+    /**
+     * Set the API base URL.
+     *
+     * @see Builder#baseUrl(String)
+     */
+    public ArmeriaRetrofitBuilder baseUrl(String baseUrl) {
+        requireNonNull(baseUrl);
+        String path = URI.create(baseUrl).getPath();
+        if (!path.isEmpty() && !SLASH.equals(path.substring(path.length() - 1))) {
+            throw new IllegalArgumentException("baseUrl must end in /: " + baseUrl);
+        }
+        this.baseUrl = baseUrl;
+        return this;
+    }
+
+    /**
+     * Sets a function for creating client with custom {@link ClientOption}s.
+     *
+     * @param configurator a {@link Function} whose input is the original {@link ClientOptions} of the client
+     *                     being derived from and whose output is the {@link ClientOptions} of the new derived
+     *                     client
+     */
+    public ArmeriaRetrofitBuilder withClientOptions(
+            Function<? super ClientOptions, ClientOptions> configurator) {
+        requireNonNull(configurator);
```

Comment:
Global comment: Please specify the parameter name when using `requireNonNull()`: e.g. `requireNonNull(configurator, "configurator")` If you import the settings as mentioned [here](https://github.com/line/armeria/blob/master/CONTRIBUTING.md#intellij-idea), you'll be able to use the template `rnn` and `rnna` which are very convenient.